src/modules/auth/components/EnterPasswordForm/EnterPasswordForm.js

```diff
@@ -27,10 +24,9 @@ const EnterPasswordForm = ({ onEnterPasswordSuccess, title, encryptedAccount, is
     formState: { errors },
   } = useForm();
   const apiError = errors[API_ERROR_NAME];
-  const activeAccount = useSelector(selectActiveTokenAccount);
   const { getAccountByAddress } = useAccounts();
   const [currentAccount] = useCurrentAccount();
-  const requestedAccount = getAccountByAddress(activeAccount.summary?.address);
+  const requestedAccount = getAccountByAddress(currentAccount.metadata.address);
```

Comment:
Why this change?  This component is used when adding an account by recovery file. Meaning that currentAccount can be empty for that usecase. Removing the `?` will make it crash, needs to use `currentAccount.metadata?.address` or more refactor needed

Reply:
True its should be use account from the next line
```suggestion
  const requestedAccount = getAccountByAddress(account.metadata.address);
```

Reply:
I see so just a removal of "dead" code. Then we still need optional chaining on the `currentAccount` to not break the "add an account by recovery file" function.

In the add an account by recovery file we pass in the `encryptedAccount` prop and the whole `currentAccount` logic is dead in that case. 

It would be better if this component always took in the correct account and avoided this logic ->
```
  const account = useMemo(
    () => encryptedAccount || requestedAccount || currentAccount,
    [currentAccount]
  );
```
to make it more understandable. But it might be too big of a refactor right now?

Reply:
yhea so the idea is that currentAccount would always have metadata so far the user is logged and since for any use case that a password is to be provided the user is logged in, `currentAccount.metadata` would never be empty.

Reply:
@eniolam1000752 
This `EnterPasswordForm` can currently be used when a user is not completely loged in, for example the "Add an account by recovery file" flow and will in that case be empty. 

The "Add an account by recovery file" flow looks like this:

1. Open a new private window
2. Go to http://0.0.0.0:8080/#/account/add/by-file
3. Upload a file and continue
4. Now `EnterPasswordForm` is used even though you are not loged in. And to prevent a crash here we need to have the optional chaining.

Reply:
hmmmm 🤔 true